composer/datasets/in_context_learning_evaluation.py

```diff
@@ -135,16 +405,35 @@ def get_lm_task_dataloader(dataset_uri: str, tokenizer: Union[transformers.PreTr
         batch_size (int): Size of a batch used for eval
         max_seq_len (int): The sequence length expected by the model
         eos_tok_id (int): The special token reserved for padding the ends of batches
+        num_fewshot (int): The number of complete fewshot examples to pad each test example with
+        prompt_string (str): Prompt string to put once before all fewshot examples/test examples (e.g. 'translate english to french')
+        example_delimiter (str): Separator that goes between individual examples (e.g. '\n')
+        continuation_delimiter: (str): Separator that goes between context and continuation in each example (e.g. '->')
 
     Returns:
         DataLoader: A dataloader used for performing in-context learning evaluation on the dataset provided.
     """
-    dataset = InContextLearningLMTaskDataset(dataset_uri, tokenizer, max_seq_len, eos_tok_id)
-    sampler = dist.get_sampler(dataset, drop_last=False, shuffle=True)
+
+    if icl_task_type == 'multiple_choice':
+        dataset = InContextLearningMultipleChoiceTaskDataset(dataset_uri, tokenizer, max_seq_len, eos_tok_id,
+                                                             num_fewshot, prompt_string, example_delimiter,
+                                                             continuation_delimiter)
+        batch_size = max(dataset.num_choices, batch_size)
+        effective_batchsize = batch_size // dataset.num_choices
+    elif icl_task_type == 'language_modeling':
+        dataset = InContextLearningLMTaskDataset(dataset_uri, tokenizer, max_seq_len, eos_tok_id, num_fewshot,
+                                                 prompt_string, example_delimiter, continuation_delimiter)
+        effective_batchsize = batch_size
+    else:
+        raise Exception(f'Unrecognized ICL task type: {icl_task_type}')
+
+    sampler = dist.get_sampler(dataset, drop_last=False, shuffle=False)
+
     return DataSpec(DataLoader(
         dataset,
-        batch_size=batch_size,
+        batch_size=effective_batchsize,
         sampler=sampler,
         collate_fn=dataset.collate_fn,
     ),
+                    device_transforms=None,
                     get_num_samples_in_batch=dataset.get_num_samples_in_batch)
```

Comment:
Related to above, if we can get away with Composer's default `get_num_samples_in_batch` function, then we can skip the `DataSpec` wrapper entirely and just return a `DataLoader`. But ofc if the custom function is needed, then no changes here.